Tests fail: cannot find test program files on Windows in the build tree with a VS generator

**Describe the bug**
Additional ctest (all of them) failed on windows

**To Reproduce**
Steps to reproduce the behavior:

Open x64 Native Tools Command Prompt for VS 2022
```
git clone https://github.com/open-quantum-safe/liboqs.git
cd liboqs
mkdir build && cd build
cmake ..
msbuild ALL_BUILD.vcxproj
msbuild tests\run_tests.vcxproj
```

**Expected behavior**
A clear and concise description of what you expected to happen.
All test cases pass

**Environment (please complete the following information):**
 - OS: Windows 10 x64 1903
 - OpenSSL version: 3.1.2
 - Compiler version used: MSVC v143/Visual Studio 2022-ver 17.7.4
 - liboqs version: main branch

**Additional context**
VS log for Debug/x64 build

```
......
1>program_name = 'speed_sig'
1>
1>    def path_to_executable(program_name):
1>        path = "."
1>        if sys.platform.startswith("win"):
1>            if 'APPVEYOR_BUILD_FOLDER' not in os.environ: os.environ['APPVEYOR_BUILD_FOLDER'] = "."
1>            path = os.path.join(path, os.environ['APPVEYOR_BUILD_FOLDER'])
1>        path = os.path.join(path, get_current_build_dir_name(), "tests")
1>        for executable in [
1>            os.path.join(path, program_name),
1>            os.path.join(path, program_name + ".EXE"),
1>            os.path.join(path, program_name + ".exe")]:
1>                if os.path.isfile(executable):
1>                    return executable1>>       assert False, "Unable to find executable file {}".format(program_name)
1>E       AssertionError: Unable to find executable file speed_sig
1>
1>tests\helpers.py:150: AssertionError
1>=========================== short test summary info ===========================
1>FAILED tests/test_cmdline.py::test_sig[Dilithium5] - AssertionError: Unable t...
1>FAILED tests/test_alg_info.py::test_alg_info_kem[FrodoKEM-1344-SHAKE] - Asser...
1>FAILED tests/test_cmdline.py::test_kem[Classic-McEliece-348864] - AssertionEr...
1>FAILED tests/test_alg_info.py::test_alg_info_kem[Classic-McEliece-348864] - A...
1>FAILED tests/test_alg_info.py::test_alg_info_sig[Dilithium2] - AssertionError...
......
```

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -113,6 +113,18 @@ endif()
 get_property(OQS_COMPILE_OPTIONS TARGET speed_kem PROPERTY COMPILE_OPTIONS)
 add_definitions(-DOQS_COMPILE_OPTIONS="[${OQS_COMPILE_OPTIONS}]")
 
+if (CMAKE_GENERATOR MATCHES "Visual Studio")
+    # With Visual studio the output of tests go into a folder with the configuration option. Force it to the same folder as if
+    # generating with Ninja
+    set_target_properties(
+        dump_alg_info example_kem kat_kem test_kem example_sig kat_sig test_sig test_sig_mem test_kem_mem speed_kem speed_sig
+        PROPERTIES 
+        RUNTIME_OUTPUT_DIRECTORY_DEBUG          "${CMAKE_BINARY_DIR}/tests"
+        RUNTIME_OUTPUT_DIRECTORY_RELEASE        "${CMAKE_BINARY_DIR}/tests"
+        RUNTIME_OUTPUT_DIRECTORY_RELWITHDEBINFO "${CMAKE_BINARY_DIR}/tests"
+        RUNTIME_OUTPUT_DIRECTORY_MINSIZEREL     "${CMAKE_BINARY_DIR}/tests")
+endif()
+
 # TODO: Get CMake to find python.
 # and set PATH variable in Windows
 # for DLL builds.

--- a/tests/helpers.py
+++ b/tests/helpers.py
@@ -141,9 +141,6 @@ def path_to_executable(program_name):
         if 'APPVEYOR_BUILD_FOLDER' not in os.environ: os.environ['APPVEYOR_BUILD_FOLDER'] = "."
         path = os.path.join(path, os.environ['APPVEYOR_BUILD_FOLDER'])
     path = os.path.join(path, get_current_build_dir_name(), "tests")
-    if sys.platform.startswith("win"):
-        if 'GITHUB_ACTIONS' in os.environ:
-            path = os.path.join(path, "Debug")
     for executable in [
         os.path.join(path, program_name),
         os.path.join(path, program_name + ".EXE"),

